tests/ui/option_if_let_else.stderr

```diff
@@ -188,14 +188,32 @@ LL +         true
 LL +     })
    |
 
+error: use Option::map_or_else instead of an if let/else
```

Comment:
issue: this lint says Option, and has Option in its name, we should probably not be linting here?

Reply:
I guess we already lint on Result here. 

Tbh I'm not convinced of the value of this lint overall, we should probably have a team discussion about it, but separately to that we shoudl figure out if it should also handle Result in the same lint (and if so, it needs to be updated)

Reply:
That's a separate issue from fixing what currently exists.